debian/patches/0064-backport-enhanced-vt.patch

```diff
@@ -0,0 +1,1925 @@
+     CFMutableDictionaryRef attributes, iosurface_properties;
++    AVVTFramesContext *hw_ctx = ctx->hwctx;
+
+     attributes = CFDictionaryCreateMutable(
+         NULL,
+@@ -173,7 +189,7 @@
+         &kCFTypeDictionaryKeyCallBacks,
+         &kCFTypeDictionaryValueCallBacks);
+
+-    cv_pixfmt = av_map_videotoolbox_format_from_pixfmt(ctx->sw_format);
++    cv_pixfmt = vt_format_from_pixfmt(ctx->sw_format, hw_ctx->color_range);
+     pixfmt = CFNumberCreate(NULL, kCFNumberSInt32Type, &cv_pixfmt);
+     CFDictionarySetValue(
+         attributes,
+@@ -267,7 +283,7 @@
+
+     if (!ctx->pool) {
+         ctx->internal->pool_internal = av_buffer_pool_init2(
+-                sizeof(CVPixelBufferRef), ctx, vt_pool_alloc_buffer, NULL);
++            sizeof(CVPixelBufferRef), ctx, vt_pool_alloc_buffer, NULL);
+         if (!ctx->internal->pool_internal)
+             return AVERROR(ENOMEM);
+     }
+@@ -326,16 +342,16 @@
+         return 0;
+
+     av_reduce(&avpar.num, &avpar.den,
+-                avpar.num, avpar.den,
+-                0xFFFFFFFF);
++              avpar.num, avpar.den,
++              0xFFFFFFFF);
+
+     num = CFNumberCreate(kCFAllocatorDefault,
+-                            kCFNumberIntType,
+-                            &avpar.num);
++                         kCFNumberIntType,
++                         &avpar.num);
+
+     den = CFNumberCreate(kCFAllocatorDefault,
+-                            kCFNumberIntType,
+-                            &avpar.den);
++                         kCFNumberIntType,
++                         &avpar.den);
+
+     par = CFDictionaryCreateMutable(kCFAllocatorDefault,
+                                     2,
+@@ -375,20 +391,20 @@
+ CFStringRef av_map_videotoolbox_chroma_loc_from_av(enum AVChromaLocation loc)
+ {
+     switch (loc) {
+-    case AVCHROMA_LOC_LEFT:
+-        return kCVImageBufferChromaLocation_Left;
+-    case AVCHROMA_LOC_CENTER:
+-        return kCVImageBufferChromaLocation_Center;
+-    case AVCHROMA_LOC_TOP:
+-        return kCVImageBufferChromaLocation_Top;
+-    case AVCHROMA_LOC_BOTTOM:
+-        return kCVImageBufferChromaLocation_Bottom;
+-    case AVCHROMA_LOC_TOPLEFT:
+-        return kCVImageBufferChromaLocation_TopLeft;
+-    case AVCHROMA_LOC_BOTTOMLEFT:
+-        return kCVImageBufferChromaLocation_BottomLeft;
+-    default:
+-        return NULL;
++        case AVCHROMA_LOC_LEFT:
++            return kCVImageBufferChromaLocation_Left;
++        case AVCHROMA_LOC_CENTER:
++            return kCVImageBufferChromaLocation_Center;
++        case AVCHROMA_LOC_TOP:
++            return kCVImageBufferChromaLocation_Top;
++        case AVCHROMA_LOC_BOTTOM:
++            return kCVImageBufferChromaLocation_Bottom;
++        case AVCHROMA_LOC_TOPLEFT:
++            return kCVImageBufferChromaLocation_TopLeft;
++        case AVCHROMA_LOC_BOTTOMLEFT:
++            return kCVImageBufferChromaLocation_BottomLeft;
++        default:
++            return NULL;
```

Comment:
Please avoid cosmetic changes in backports, such as only fixing indentation, to keep the patch as slim as possible. I've seen changes like this in other files in this patch as well.

Reply:
This will happen if the upstream has cosmetic changes and I just put them as-is.

So I'm supposed to revert these changes?

Reply:
If it was the upstream who made cosmetic changes then there is no problem to backport them.
But for this function, it has not changed [since FF 5.0](https://github.com/FFmpeg/FFmpeg/blame/ef917950f0298a812bebfed2443626972a7d8f29/libavutil/hwcontext_videotoolbox.c#L391-L409). So it's best to just restore them to default.